nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -2000,12 +2010,16 @@ public File getTemplateConf(String template) throws ConfigurationException {
 
     /**
      * Gets the Java options with 'nuxeo.*' properties substituted. It enables usage of property like ${nuxeo.log.dir}
-     * inside JAVA_OPTS.
+     * inside JAVA_OPTS. The default values are defined by {@link #JAVA_OPTS_DEFAULT}.
      *
-     * @return the java options string.
-     */
-    protected String getJavaOpts(String key, String value) {
-        return StrSubstitutor.replace(System.getProperty(key, value), getUserConfig());
+     * @return the Java options string.
+     * @see #JAVA_OPTS_DEFAULT
+     */
+    protected String getJavaOpts() {
+        String options = System.getProperty(JAVA_OPTS_PROP, JAVA_OPTS_DEFAULT);
+        return Arrays.stream(options.split("[ ]+(?=([^\"]*\"[^\"]*\")*[^\"]*$)"))
```

Comment:
Please pre-compile the regexp into a constant Pattern (and document what it parses), and use it with `THE_PATTERN.split(options)`

Reply:
and factorize

Reply:
> Please pre-compile the regexp into a constant Pattern

+1

> and document what it parses

I didn't invent this regex. This is indeed created by @jcarsique in commit dea3e8c3f4f111087ba45533694473889c36bfbb 2 years ago. I propose: _"Java OPTS split on spaces followed by an even number of quotes (or zero)"_, which is the original Javadoc in that commit. What do you think?